src/plugins/layout/shadowdom.ts

```diff
@@ -88,7 +92,11 @@ export class ShadowDom {
 
   public updateShadowNode(index: number, newLayout?: ILayoutState) {
     let shadowNode = this.getShadowNode(index);
-    assert(!!shadowNode, "updateShadowNode");
+
+    assert(!!shadowNode, "updateShadowNode", "shadowNode is missing");
```

Comment:
Discussed this offline:
I totally agree that for proper debugging richer and more robust information is needed, so i'll keep working on that, but in a separate PR. For now, we'll go with a plain yes/no check to find the bottleneck of the observed problems.

---
In reply to: [119988311](https://github.com/Microsoft/clarity-js/pull/16#discussion_r119988311) [](ancestors = 119988311)